drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,256 @@
+>= 3     branch_id       format branch id            uint32
+>= 1     in_count        varint                      1-9 bytes
+>= 1     tx_inputs       list of inputs              vector
+>= 1     out_count       varint                      1-9 bytes
+>= 1     tx_outputs      list of outputs             vector
+>= 1     lock_time       block height or timestamp   uint32
+>= 3     expiry_height   block height                uint32
+>= 2     nJoinSplit      varint                      1-9 bytes
+>= 2     vJoinSplit      list of joinsplits          vector
+>= 2     joinSplitPubKey joinSplitSig public key     32 bytes
+>= 2     joinSplitSig    signature                   64 bytes
+======== =============== =========================== =======
+
+
+Header Field
+------------
+
+The first four bytes of pre-Overwinter and Overwinter transactions are little-endian encoded.
+
+Version 1 transaction (txid 5c6ba844e1ca1c8083cd53e29971bd82f1f9eea1f86c1763a22dd4ca183ae061 https://zcash.blockexplorer.com/tx/5c6ba844e1ca1c8083cd53e29971bd82f1f9eea1f86c1763a22dd4ca183ae061)
+
+* begins with little-endian byte sequence [0x01, 0x00, 0x00, 0x00]
+* deserialized as 32-bit signed integer with decimal value of 1
+
+Version 2 transaction (txid 4435bf8064e74f01262cb1725fd9b53e600fa285950163fd961bed3a64260d8b https://zcash.blockexplorer.com/tx/4435bf8064e74f01262cb1725fd9b53e600fa285950163fd961bed3a64260d8b)
+
+* begins with little-endian byte sequence [0x02, 0x00, 0x00, 0x00]
+* deserialized as 32-bit signed integer with decimal value of 2
+
+Transaction parsers for versions of Zcash prior to Overwinter, and for most other Bitcoin forks, require the transaction version number to be positive.
+
+With the version 3 transaction format, the first four bytes of a serialized transaction, the 32-bit header, are made up of two fields as shown in the able above:
+
+* 1 bit Overwinter flag, must be set
+* 31 bit unsigned int for the Version
+
+Pre-Overwinter parsers will deserialize these four bytes as a 32-bit signed integer.  With two's complement integers, the most significant bit indicates whether an integer is positive or negative.  With the Overwinter flag set, the transaction version will be negative, resulting in pre-Overwinter parsers rejecting the transaction as invalid.  This provides transaction replay protection between per-Overwinter and Overwinter software.
+
+Consider the following example of a serialized version 3 transaction.
+
+Pre-Overwinter parser:
+
+* data begins with little-endian byte sequence: [0x03, 0x00, 0x00, 0x80]
+* deserialized as 32-bit signed integer
+
+  * with hexadecimal value of 0x80000003 (most significant bit is set)
+  * decimal value of -2147483645
+
+Legacy parsers will expect the version to be a positive value, such as 1 or 2, and will thus reject the Overwinter transaction as invalid.
+
+Overwinter parser:
+
+- data begins with little-endian byte sequence: [0x03, 0x00, 0x00, 0x80]
+- deserialized as 32-bit unsigned integer
```

Comment:
seems a little awkward saying you treat the 32 bits as an integer, when you use one of the bits as a separate flag.
Also, current phrasing leaves a little doubt whether the 31 bit version is treated as signed or unsigned.
Perhaps just say "32 bits decomposed as.. "
..
and then
- version (bits 30 - bit 0)  deserialized as an unsigned integer have a decimal value of 3